lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -632,6 +640,77 @@ def complete(self, dataset: Dataset, params: ApiParams, summary: JSON) -> None:
         """
         pass
 
+    def _analyze_bulk(self, results: Iterator[tuple[bool, Any]]) -> BulkResults:
+        """Elasticsearch returns one response result per action. Each is a
+        JSON document where the first-level key is the action name
+        ("update", "delete", etc.) and the value of that key includes the
+        action's "status", "_index", etc; and, on failure, an "error" key
+        the value of which gives the type and reason for the failure.
+
+        We assume there will be a single first-level key corresponding to
+        the action generated by the subclass and we use that without any
+        validation to access the status information.
+
+        Internally report a summary of successes and Elasticsearch failure
+        reasons: this will look something like
+
+        {
+          "ok": {
+            "index1": 1,
+            "index2": 500,
+            ...
+          },
+          "elasticsearch failure reason 1": {
+            "index2": 5,
+            "index5": 10
+            ...
+          },
+          "elasticsearch failure reason 2": {
+            "index3": 2,
+            "index4": 15
+            ...
+          }
+        }
+        """
+        report = defaultdict(dict)
```

Comment:
That's actually kinda cute. I didn't do that originally because I was trying to hide the count and error count inside the `report` dict, and *those* would be top-level `int` vs the `dict.int` structure of the other top-level entries. However since I backed off on that idea for other reasons ... yeah, I could. Is it worth throwing away three approvals? Maybe not ... 😆